Acs 7862 shared qlty config

rebeccafae opened the pull request:
**_JIRA Card_**: https://qcentrix.atlassian.net/browse/ACS-7862
**_Summary and description of changes_**:
create a shared base config

See an example set up here: https://github.com/q-centrix/web/pull/9590


  
### How to set this up on other repos

- run `qlty init`, that will make your qlty directory and your needed files; you may need to [install qlty](https://docs.qlty.sh/cli/quickstart)
- update the qlty.toml to look like the following
```
config_version = "0"

[[source]]
name = "default"
default = true

[[source]]
name = "Q-Centrix"
repository = "https://github.com/q-centrix/style-guides/"
branch = "main"

[[plugin]]
name = "rubocop"
version = "1.65.1"
package_file = "Gemfile"
package_filters = ["rubocop"]

[[plugin.fetch]]
url = "https://raw.githubusercontent.com/q-centrix/style-guides/main/ruby/.rubocop-1-60-all.yml"
path = ".rubocop-https---raw-githubusercontent-com-q-centrix-style-guides-main-ruby--rubocop-1-60-all-yml"
```

- Done! You can also use the qlty cli now, for example `qlty check` should run the exact same as what it will do on your PR

rebeccafae committed:
ACS-7862 first draft

rebeccafae committed:
does the location matter?

rebeccafae committed:
which file is it using?

rebeccafae committed:
remove top level file

rebeccafae committed:
remove all plugins except one

rebeccafae committed:
move to root

rebeccafae committed:
match current toml used by web repo

rebeccafae committed:
update deprecation

rebeccafae committed:
add a file taht doesn't exist to confirm nothing breaks

rebeccafae committed:
undo test

stephbegle commented:
Nice, thanks for making this @rebeccafae, I added my questions/comments on the confluence doc!

rebeccafae committed:
Merge remote-tracking branch 'origin/main' into ACS-7862-shared-qlty-config

rebeccafae committed:
remove rubocop from source.toml

- it will be specified in qlty.toml so repos can change version when they want to

albertski reviewed (approved):
Looks great.  I went through all the plugins again and they all seem to make sense for us.
